Maple2.Database/Storage/Game/GameStorage.Market.cs

```diff
@@ -54,5 +59,109 @@ public partial class Request {
+                        continue;
+                    }
+                    premium.AdditionalQuantities = Context.PremiumMarketItem
+                        .Select(ToMarketEntry)
+                        .Where(subEntry => subEntry != null &&
+                                           subEntry.ParentId == premium.Id)
+                        .ToList()!;
+                }
+            } else {
+                results = Context.PremiumMarketItem
+                    .AsEnumerable()
+                    .Select(ToMarketEntry)
+                    .Where(entry => entry != null &&
+                                    entry.ParentId == 0 &&
+                                    tabIds.Contains(entry.TabId) &&
+                                    (entry.ItemMetadata.Name != null && entry.ItemMetadata.Name.Contains(searchString, StringComparison.OrdinalIgnoreCase)) &&
+                                    gender.HasFlag(entry.ItemMetadata.Limit.Gender.Flag()) &&
+                                    (entry.ItemMetadata.Limit.JobLimits.Length == 0 || entry.ItemMetadata.Limit.JobLimits.Any(jobs => job.Code().Any(codes => codes == jobs)))
+                    )
+                    .ToList()!;
+
+                // Get any additional quantities
+                foreach (MarketItem marketEntry in results) {
+                    if (marketEntry is not PremiumMarketItem premium) {
+                        continue;
+                    }
+                    premium.AdditionalQuantities = Context.PremiumMarketItem
+                        .Select(ToMarketEntry)
+                        .Where(subEntry => subEntry != null && subEntry.ParentId == premium.Id)
+                        .ToList()!;
+                }
+            }
+
+
+            if (sortGender) {
+                results = results.OrderBy(item => item.ItemMetadata.Limit.Gender);
+            }
+            if (sortJob) {
+                results = results.OrderBy(item => item.ItemMetadata.Limit.JobLimits);
+            }
+            return results.ToList();
+        }
+
+        public ICollection<MarketItem> GetMarketItems(MeretMarketSection section, int tabId) {
+            IEnumerable<MarketItem> results = new List<MarketItem>();
+            switch (section) {
+                case MeretMarketSection.All:
+                    results = Context.PremiumMarketItem
+                        .AsEnumerable()
+                        .Select(ToMarketEntry)
+                        .Where(entry => entry != null &&
+                                        entry.ParentId == 0 &&
+                                        entry.TabId == tabId)
+                        .ToList()!;
+                    break;
+            }
+
+            foreach (MarketItem marketEntry in results) {
+                if (marketEntry is not PremiumMarketItem premium) {
+                    continue;
+                }
+                premium.AdditionalQuantities = Context.PremiumMarketItem
```

Comment:
Very bad, this seems like a full table scan.
`Context.PremiumMarketItem.Select(ToMarketEntry)`

Reply:
I'm not sure what you mean? Isn't that what I'm doing? I would have to look through the entire table for it.

Reply:
So the way SQL works is there is an engine that can execute queries on the DB server before returning results. Ideally you limit the amount of data returned to you because it's much more efficient to let the DB engine handle that part.

In this case, you would do something like:
```cs
Context.PremiumMarketItem
    .Where(subEntry => subEntry != null && subEntry.ParentId == premium.Id) // On DB
    .Select(ToMarketEntry) // Local
    .ToList()!; // Local
```

Reply:
Makes sense. I pushed a few updates to remove some repeated code and fix this. hopefully.